From 138e850fab106b5235178848b3e0d33e25f4d3a2 Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Wed, 2 Mar 2016 17:52:23 -0500
Subject: [PATCH] keytab-lilo: update to support kbd 2.0.3 format

Since kbd 2.0.3 (commit 6ff47cf2) the format generated by 'loadkeys -m' has
changed slightly, using "unsigned short" instead of "u_short"

Signed-off-by: Olivier Brunel <jjk@jjacky.com>
---
 utils/keytab-lilo | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/utils/keytab-lilo b/utils/keytab-lilo
index 9e34160..337a869 100755
--- a/utils/keytab-lilo
+++ b/utils/keytab-lilo
@@ -48,9 +48,9 @@ sub load_map
     $empty = 1;
     while (<FILE>) {
 	chop;
-	if (/^(static\s+)?u_short\s+(\S+)_map\[\S*\]\s+=\s+{\s*$/) {
+	if (/^(static\s+)?(u_|unsigned )short\s+(\S+)_map\[\S*\]\s+=\s+{\s*$/) {
 	    die "active at beginning of map" if defined $current;
-	    $current = $pfx.":".$2;
+	    $current = $pfx.":".$3;
 	    next;
 	}
 	undef $current if /^};\s*$/;
-- 
2.7.4.GIT

